lms/lib/comment_client/comment.py

```diff
@@ -32,10 +32,10 @@ def thread(self):
 
     @classmethod
     def url_for_comments(cls, params={}):
-        if params.get('thread_id'):
-            return _url_for_thread_comments(params['thread_id'])
-        else:
+        if params.get('parent_id'):
             return _url_for_comment(params['parent_id'])
+        else:
+            return _url_for_thread_comments(params['thread_id'])
```

Comment:
If both a thread_id and a parent_id are specified, then we should create the comment as a child of the specified parent. The previous code would create the comment as a child of the thread.